File: android/api-surface-check/src/negativeCheckShowTransition/kotlin/jp/kamusoft/ksdialogs/apicheck/RejectsTransitionArgumentOnShow.kt
package jp.kamusoft.ksdialogs.apicheck

import jp.kamusoft.ksdialogs.DialogResult
import jp.kamusoft.ksdialogs.DialogTransition
import jp.kamusoft.ksdialogs.KsDialog

/**
 * show に transition 引数はない (出入りの演出は添付でのみ供給する。core/ADR-0017)。
 *
 * このソースは `-Pksdialogs.negativeCheck.showTransition` を付けたときだけビルドに加わり、
 * **コンパイルエラーで失敗すること**が期待結果になる。
 */
public object RejectsTransitionArgumentOnShow {
    public suspend fun show(dialogs: KsDialog): DialogResult<Boolean> =
        dialogs.show(ConsumerDialogViewModel("こんにちは"), transition = DialogTransition.none())
}

File: maui/KsDialogs.Maui.ApiSurfaceCheck/NegativeChecks/RejectsTransitionArgumentOnShow.cs
using System.Threading.Tasks;

namespace KsDialogs.ApiSurfaceCheck.NegativeChecks;

/// <summary>show に transition 引数はない (出入りの演出は添付でのみ供給する。core/ADR-0017)。</summary>
/// <remarks>
/// このソースはビルドプロパティ KsDialogsNegativeCheckShowTransition を true にしたときだけ加わり、
/// <b>コンパイルエラーで失敗すること</b>が期待結果になる。
/// 期待する診断: CS1739 (transition という名前のパラメーターがない)
/// </remarks>
internal static class RejectsTransitionArgumentOnShow
{
    public static Task<DialogResult<bool>> Show(IKsDialog dialogs) =>
        dialogs.ShowAsync(new ConsumerDialogViewModel(), transition: DialogTransition.None());
}
